### ISSUE TITLE
Enable event detection

### ISSUE THREAD

**Issue by sdwfrost** [rveltz/LSODA.jl#26]
This issue is mostly to @ChrisRackauckas ; I'd like to use [event detection](http://docs.juliadiffeq.org/latest/features/callback_functions.html) in LSODA.jl; I assume that this will require some additions to liblsoda in order to support returning from lsoda every timestep?

**Comment by rveltz** [comment]
I guess so but you checked that it is not saved in `lsoda_common_t `

**Comment by sdwfrost** [comment]
One can always do event detection after running a system between [0,t], but for efficiency, it's probably best to check the condition each time step, then interpolate only between two consecutive timesteps.

**Comment by rveltz** [comment]
Yes. I am not very keen on event detection but one can probably hack it. Did yu check  lsoda_common_t?

**Comment by sdwfrost** [comment]
Yes, but one should probably have an interface to the one-step solver [stoda](https://github.com/sdwfrost/liblsoda/blob/master/src/stoda.c) in order to facilitate the common interface for event handling.

**Comment by ChrisRackauckas** [comment]
>Yes, but one should probably have an interface to the one-step solver stoda in order to facilitate the common interface for event handling.

Well, what we have here is currently enough since it's already controlled a stepwise level and I believe the GC works now? What stopped Sundials is that, in order for the GC to register the size, it needs a custom allocator, and so setting up the `integrator` interface is unviable since we cannot rely on the GC there. Here, we can just wrap it all in a type which has the `integrator` interface, and then the event handling stuff is almost free.

My plan there is to just take what's in the `*DiffEq` packages and slap that on these interop packages. The reasoning is here:

https://github.com/JuliaDiffEq/Sundials.jl/issues/109#issuecomment-289245548

The rootfinding algorithm that comes in "the standard timestepping packages" isn't very good... and since we have Julia we can rely on the fact that our native methods are just as fast and put something more robust there which just uses the already existing interval interpolations.

That said... I need to do my advancement right now so I won't be able to do this for awhile.

**Comment by ChrisRackauckas** [comment]
Is the call to the intermediate interpolation wrapped?
